src/ctap/data_formats.rs

```diff
@@ -445,66 +445,93 @@ pub struct PublicKeyCredentialSource {
     pub user_handle: Vec<u8>, // not optional, but nullable
     pub other_ui: Option<String>,
     pub cred_random: Option<Vec<u8>>,
+
+    /// Contains the unknown fields when parsing a CBOR value.
+    ///
+    /// Those fields could either be deleted fields from older versions (they should have reserved
+    /// tags) or fields from newer versions (the tags should not be reserved). If this is empty,
+    /// then the parsed credential is probably from the same version (but not necessarily).
+    pub unknown_fields: BTreeMap<cbor::KeyType, cbor::Value>,
+}
+
+// We serialize credentials for the persistent storage using CBOR maps. Each field of a credential
+// is associated with a unique tag, implemented with a CBOR unsigned key.
+#[repr(u64)]
+enum PublicKeyCredentialSourceField {
+    CredentialId = 0,
+    PrivateKey = 1,
+    RpId = 2,
+    UserHandle = 3,
+    OtherUi = 4,
+    CredRandom = 5,
+    // When a field is removed, its tag should be reserved and not used for new fields. We document
+    // those reserved tags below.
+    // Reserved tags: none.
+}
+
+impl From<PublicKeyCredentialSourceField> for cbor::KeyType {
+    fn from(field: PublicKeyCredentialSourceField) -> cbor::KeyType {
+        (field as u64).into()
```

Comment:
Currently, the enums like `SignatureAlgorithm` do not implement
`impl From<SignatureAlgorithm> for cbor::Value`
or
`impl From<SignatureAlgorithm> for cbor::KeyType`
since the cast is a simple `as i64` or `as u64`.

My comment is about consistency. I could these kinds of conversions for other enums with integer representations if we start adding them here.

Reply:
Syntactical consistency should not break design (and should also not break engineering and small PRs). Would you use `void *` for all pointers just for consistency? Things should be implemented if they make sense, i.e. they are part of a design and not because they look consistent.

It looks to me that those consistency changes can be done in another PR if needed (and I'm not doing them because they don't look correct and well motivated). So I'm only addressing those for added/modified lines in this PR that are problematic. Please flag those explicitly.

Reply:
I didn't want to stop you from implementing this trait. Just mentioned that I did it differently in this file so far. I think implementing the `From` trait for all of my enums is the way to go. This comment is not actionable, but more informative (like the one below about consistency). Feel free to resolve.

Reply:
I see. It wasn't clear to me. Thanks for clarifying!

Note that I'm reluctant modifying the CTAP structs because they are specified by FIDO and as such need a good understanding of the specification which I don't have. As clarified before, the source credential is not specified and is thus much safer to touch.